parquet/src/main/java/org/apache/iceberg/parquet/ParquetTypeVisitor.java

```diff
@@ -123,19 +60,115 @@
     }
   }
 
-  private static <T> T visitField(Type field, ParquetTypeVisitor<T> visitor) {
-    visitor.fieldNames.push(field.getName());
+  private static <T> T visitList(GroupType list, ParquetTypeVisitor<T> visitor) {
```

Comment:
This was moved into its own method. The only changes are to use the new before and after methods.